packages/api/amplify_api/lib/src/method_channel_api.dart

```diff
@@ -248,21 +237,64 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
   }
 
   // ====== RestAPI ======
-  RestOperation _restFunctionHelper(
-      {required String methodName, required RestOptions restOptions}) {
-    // Send Request cancelToken to Native
-    String cancelToken = UUID.getUUID();
 
-    Future<RestResponse> futureResponse =
-        _callNativeRestMethod(methodName, cancelToken, restOptions);
+  Future<AWSStreamedHttpResponse> _restResponseHelper({
+    required String methodName,
+    required String path,
+    required String cancelToken,
+    HttpPayload? body,
+    Map<String, String>? headers,
+    Map<String, String>? queryParameters,
+    String? apiName,
+  }) async {
+    Uint8List? bodyBytes;
```

Comment:
We should only be doing this for iOS/Android, not for Dart

Reply:
Yep, my understanding as well. This method basically works around the limitations of existing serialization/deserialization implementation which makes streamed request bodies difficult. In dart, can (and should) allow request body to stream without converting all at once.